flow-typed/platform.js

```diff
@@ -71,3 +71,20 @@ declare module 'react-router-dom/matchPath' {
 	declare function matchPath(pathname: string, options: MatchPathOptions): null | Match;
 	declare module.exports: typeof matchPath;
 }
+
+declare type LocationShape = {
+  pathname?: string,
```

Comment:
optional prop. Any key that is not explicitly mentioned in the type definition will throw an error